admin/src/js/services/select2-search.js

```diff
@@ -147,12 +206,43 @@ angular.module('inboxServices').factory('Select2Search',
           }
         }
 
-        return resolution.then(function() {
-          selectEl.trigger('change');
+        return resolution.then(function() { //NoSONAR
+          $timeout(() => { //NoSONAR
+            selectEl.trigger('change');
+          }, 1000);
           return selectEl;
         });
       };
 
+      const updateSelect2DataWithDoc = function (selectEl, docId, doc) {
+        const select2Data = selectEl.select2('data') || [];
+        const selected = select2Data.find((d) => d.id === docId);
+        if (selected) {
+          selected.doc = doc;
+        } else {
+          select2Data.push({
+            id: docId,
+            doc: doc,
+            text: doc.name,
+          });
+        }
+        selectEl.select2('data', select2Data);
+        $timeout(() => { //NoSONAR
+          selectEl.trigger('change');
+        }, 1000);
```

Comment:
Same. We should not have delays like this, we need to figure out what is happening here.

Reply:
Is the problem using `timeout` with a 1 sec delay or is it using `timeout` all together ? 

From my understanding of [ some research](https://stackoverflow.com/questions/12729122/angularjs-prevent-error-digest-already-in-progress-when-calling-scope-apply) and the [docs](https://docs.angularjs.org/error/$rootScope/inprog) `timeout`is one way to deal with the `Error: $rootScope:inprog
Action Already In Progress`. 

```
$timeout(() => {
          selectEl.trigger('change');
        });
```

With the event being triggered outside AngularJS's digest cycle by JQuery, I don't see a simpler workaround. But I am open to suggestions.

Reply:
Using $timeout is ok. Using $timeout with a delay is not ok.